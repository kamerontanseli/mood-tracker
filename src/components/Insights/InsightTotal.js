import React from "react";
import Progress from "../general/Progress";
import Header2 from "../general/Header2";
import { getMood } from "../../utils/string";
import { insightAverage } from "../../utils/array";

const InsightTotal = ({ insights }) => (
  <React.Fragment>
    <Progress value={insightAverage(insights)} max={7} />
    <Header2>
      Your average mood is <strong>{getMood(Math.round(insightAverage(insights)))}</strong>
    </Header2>
  </React.Fragment>
);

export default InsightTotal;
