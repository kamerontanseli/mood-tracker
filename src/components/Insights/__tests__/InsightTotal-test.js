import React from "react";
import renderer from "react-test-renderer";
import InsightTotal from "../InsightTotal";

describe("render", () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <InsightTotal insights={[{ mood: 1 }, { mood: 2 }, { mood: 3 }]} />
    );
    expect(snapshot).toMatchSnapshot();
  });
});
