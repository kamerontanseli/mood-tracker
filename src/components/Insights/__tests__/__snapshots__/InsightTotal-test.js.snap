// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
Array [
  <progress
    className="Progress"
    max={7}
    value={2}
  />,
  <h2
    className="Header2"
  >
    Your average mood is 
    <strong>
      Bad
    </strong>
  </h2>,
]
`;
